Controllers/ChatController.cs

```diff
@@ -0,0 +1,135 @@
+        [HttpPost("consultant")]
+        public async Task<IActionResult> ConsultantChat(ConsultantChatConnection consultantChatConnection)
+        {
+            var conversation = await _chatRepository.GetConversationByIdAsync(consultantChatConnection.ConversationId);
+            var consultantId = User.FindFirst(ClaimTypes.NameIdentifier).Value;
+
+            if (
+                conversation.Status == ConversationStatus.IN_PROGRESS &&
+                conversation.ConsultantId == consultantId
+                ) return View("ConsultantChat", new ConsultantChatViewModel
+                {
+                    ConversationId = consultantChatConnection.ConversationId,
+                    RedirectAction = consultantChatConnection.RedirectAction,
+                    Host = conversation.Host,
+                    Path = conversation.Path
+                });
+
+            if (conversation.Status != ConversationStatus.NEW) return RedirectToAction("NewConversationList", new ModalViewModel
+            {
+                ModalLabel = "Already taken",
+                ModalText = new List<string>() { "Other consultant has already joined to this conversation." },
+                IsVisible = true,
+                ModalType = ModalStyles.ERROR
+            });
+
+            var isSaved = await _chatRepository.AssignConsultantToConversation(conversation,
+                                                                             consultantId,
+                                                                             consultantChatConnection.RowVersion);
+
+            if (!isSaved) return RedirectToAction("NewConversationList", new ModalViewModel
+            {
+                ModalLabel = "Already taken",
+                ModalText = new List<string>() { "Other consultant has already joined to this conversation." },
+                IsVisible = true,
+                ModalType = ModalStyles.ERROR
+            });
+
+
+            return View("ConsultantChat", new ConsultantChatViewModel
+            {
+                ConversationId = consultantChatConnection.ConversationId,
+                RedirectAction = consultantChatConnection.RedirectAction,
+                Host = conversation.Host,
+                Path = conversation.Path
+            });
+        }
+
+        [Authorize(Roles = UserRoleValue.CONSULTANT)]
+        [HttpGet("new-conversation-list")]
+        public async Task<IActionResult> NewConversationList(int pageIndex = 1, ModalViewModel modalViewModel = null)
+        {
+            var newConversations = _chatRepository.GetNewConversationsQuery();
+
+            var newConversationsPaginated = await PaginatedList<Conversation>.CreateAsync(
+                                                                                newConversations,
+                                                                                pageIndex,
+                                                                                10);
```

Comment:
I would personally do something like `const int PAGE_SIZE = 10;` at the start of controller and insert it to every PaginatedList, in order to not use magic numbers